array-manipulation-1/src/compact.js

```diff
@@ -0,0 +1,21 @@
+/* exported compact */
+
+function compact(array) {
+  var validateArray = [];
+  for (var i = 0; i <= (array.length - 1); ++i) {
+    if (array[i] !== false) {
+      if (array[i] !== null) {
+        if (Number.isNaN(array[i]) === false) {
+          if (array[i] !== 0) {
+            if (array[i] !== undefined) {
+              if (array[i] !== '') {
+                validateArray.push(array[i]);
```

Comment:
This does work! But one thing it is a good idea to avoid in loops are nested conditional statements because of how much time they can take to run. You can simplify this by checking if the current value is truthy, and if so, pushing to the new array.
```javascript
if(array[i]){
validateArray.push(array[i])
}
```

Reply:
Thanks sir. Looking at it more closely, you are absolutely correct all values we were checking was falsey values and a single check would have given the output. hmm, got tricked by the testing script mirage. Thanks for all your help.